pennylane/measurements/sample.py

```diff
@@ -169,28 +169,25 @@ def shape(self, device=None):
             return tuple(
                 (shot_val,) if shot_val != 1 else tuple() for shot_val in device._raw_shot_sequence
             )
-        len_wires = len(device.wires)
+        len_wires = len(self.wires) if len(self.wires) > 0 else len(device.wires)
         return (1, device.shots) if self.obs is not None else (1, device.shots, len_wires)
 
     def _shape_new(self, device=None):
-        if device is None:
-            raise MeasurementShapeError(
-                "The device argument is required to obtain the shape of the measurement "
-                f"{self.__class__.__name__}."
-            )
-        if device.shot_vector is not None:
-            if self.obs is None:
-                return tuple(
-                    (shot_val, len(device.wires)) if shot_val != 1 else (len(device.wires),)
-                    for shot_val in device._raw_shot_sequence
-                )
-            return tuple(
-                (shot_val,) if shot_val != 1 else tuple() for shot_val in device._raw_shot_sequence
-            )
-        if self.obs is None:
-            len_wires = len(device.wires)
-            return (device.shots, len_wires) if device.shots != 1 else (len_wires,)
-        return (device.shots,) if device.shots != 1 else ()
+        num_wires = len(self.wires) if len(self.wires) > 0 else len(device.wires)
+
+        def _single_int_shape(shot_val, num_wires):
+            inner_shape = []
+            if shot_val != 1:
```

Comment:
It would be worth to add a note on why we remove the shot and wires dimension for the 1 case, it is not straightforward to see.